### PR TITLE
[FINERACT-1724] use partitions for queries with big parameter list

### PR DESCRIPTION
**Pull request by taskain7**
## Description

Fix for the PostgreSQL parameter size limitation with list partitioning

### CODE REVIEW

**Comment by adamsaghy** [comment]
@taskain7 Kindly asking you to check the failing tests!

File: fineract-provider/src/main/resources/application.properties [hunk 1]
```diff
@@ -42,6 +42,8 @@ fineract.mode.write-enabled=${FINERACT_MODE_WRITE_ENABLED:true}
 fineract.mode.batch-worker-enabled=${FINERACT_MODE_BATCH_WORKER_ENABLED:true}
 fineract.mode.batch-manager-enabled=${FINERACT_MODE_BATCH_MANAGER_ENABLED:true}
 
+fineract.query.parameter-size=${FINERACT_QUERY_PARAMETER_SIZE:65000}
```
Comment (galovics) [inline fineract-provider/src/main/resources/application.properties]: 2 things, I'd name this specifically that this is the maximum amount of parameters within an IN clause, emphasize that please.
Second thing, I'd be way more careful with the default number here and go with like 10-20k but definitely not 65k.

The thing is, DBs are often really bad at executing queries with really big IN clauses.

**Review by galovics — DISMISSED** [review]

File: fineract-provider/src/main/resources/application.properties (hunk 1 above)
Comment (adamsaghy) [reply fineract-provider/src/main/resources/application.properties]: As a default i would go with 1000 maximum.

Comment (taskain7) [reply fineract-provider/src/main/resources/application.properties]: changed it

**Review by adamsaghy — DISMISSED** [review]
LGTM

**Comment by adamsaghy** [comment]
@taskain7 Would you mind to add one more test which explicitly checks whether the partitioning is working?

**Review by galovics — APPROVED** [review]

**Review by IOhacker — COMMENTED** [review]
@taskain7 parameter limit come from the PG JDBC Driver https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/core/PGStream.java (method  sendInteger2(int val) where the upper limit is set to 65535) how this fix prevent not to reach/manual set this limit?

**Comment by taskain7** [comment]
@IOhacker 
this setting won't partition the query to separate queries that each have 65K parameter, it just returns with `PSQLException: PreparedStatement can have at most 65,535 parameters`. So we have to manually partition the query parameters to smaller chunks to make the queries executable. And also, we should respect other DBs as well.

**Pull request merged**